chore: upgrade clang-format to 1.0.32

mprobst opened the pull request:
Fixes #4596.

mprobst committed:
feat: upgrade clang-format to 1.0.32.

IgorMinar commented:
lgtm

@mprobst can you check if the test failure is a flake or what's going on there?

mprobst committed:
chore: adjust formatting to new clang-format.

- fixes wrapping for object literal keys called `template`.
- spacing in destructuring expressions.
- changes to keep trailing return types of functions closer to their
  function declaration.
- better formatting of string literals.

mary-poppins commented:
User @mprobst does not have PR merging privileges.
